hw/ip/otbn/data/otbn.hjson

```diff
@@ -349,6 +355,18 @@
           desc: "A `KEY_INVALID` error was observed."
         }
 
+        // Recoverable errors
```

Comment:
You'll also need to edit the documentation (which lists the different types of recoverable errors) and also the Python code (see `constants.py`).

Can either of these errors happen when OTBN is not running? That would be a problem because we currently rely on the fact that all "asynchronous" errors are fatal to avoid awkward race conditions.

Reply:
I think I've also adjusted the documentation (See `#design-details-list-of-errors`). Or is there a second list of errors? I've now updated `constants.py`.

About whether these errors can happen when OTBN is not running: For that to happen, OTBN would need to initiate an RND prefetch by writing the RND_PREFETCH CSR and then terminate without waiting for the prefetch to finish. This shouldn't happen under normal operation (users shouldn't end their programs like that) but only in case of fatal errors and then we probably care more about the fatal error anyway. But I am no OTBN expert. What is your view? Because we need to hold the request until it's acknowledged, maybe the correct way to handle this would be to simply ignore these two errors whenever OTBN isn't busy.

Reply:
About the documentation: you're completely right! Sorry, I missed it. The problem is that we've also got this text:
![image](https://user-images.githubusercontent.com/104845/161946446-485c5d4d-7bf5-4d29-91e4-3d5cef6bf274.png)

Before this change, all recoverable errors were caused by software bugs. Now, that's not true any more. I'm happy for us to fix the documentation as a follow-up, but we shouldn't forget to do it.

Reply:
About errors when not running: this is a bit problematic. With the current design, the only errors that can occur when not running are fatal, which means that an alert handler can see what happened by looking at the FATAL_ALERT_CAUSE register. Recoverable error causes are only reflected in ERR_BITS but this is only set at the end of an operation (and is cleared when starting the next one).

So with the current design, if we generate a recoverable error when OTBN isn't running, there's no way to get the bit out that tells you what happened.

The easy part is `RND_FIPS_CHK_FAIL`. I suggest we change it to be fatal. It shows a mis-configuration on the Ibex side and should be deterministic, so I'd just do the easy thing. (I realise that I argued for recoverable [here](https://github.com/lowRISC/opentitan/issues/8912#issuecomment-952839639), but I've seen the error of my ways!)

For `RND_REP_CHK_FAIL`, things are a bit more difficult. If we *really* think this should be a recoverable error then I guess we have to pick between
1. Ignoring the error (and throwing away the data) if it happens when we're not running
2. Adding a new RECOV_ALERT_CAUSE register that mirrors FATAL_ALERT_CAUSE but explains recoverable alerts.

Reply:
> The easy part is RND_FIPS_CHK_FAIL. I suggest we change it to be fatal. It shows a mis-configuration on the Ibex side and should be deterministic, so I'd just do the easy thing. (I realise that I argued for recoverable https://github.com/lowRISC/opentitan/issues/8912#issuecomment-952839639, but I've seen the error of my ways!)

Things like buggy OTBN code that causes a recoverable error be could said to be a 'mis-configuration on the Ibex side'. I'm not sure making this fatal is the right thing to do, as you say

>  If I understand correctly, it would be quite easy to mis-configure the CSRNG to do this wrong and I'm not sure it's worrying enough to make fatal.